### ISSUE TITLE
Better async support

### ISSUE THREAD

**Issue by barake** [landmarkhw/Dapper.GraphQL#1]
Setup a quick test project - this library works very well and is a real layer of sanity on top of graphql-dotnet. I noticed the querying is all sync, and presumably async would be better?

Looks like graphql-dotnet supports awaiting resolvers:
https://github.com/graphql-dotnet/graphql-dotnet/issues/205#issuecomment-257482356

At first glance, it looks like just `SqlQueryContext` needs an `ExecuteAsync` companion method.

Would this make sense/be within the goals of your library?

**Comment by dougrday** [comment]
I see no issues with adding support for this, async support would be beneficial.

**Comment by dougrday** [comment]
Thanks for the contribution, it needed a slight adjustment to the unit test to pass.  Looks good!

**Comment by dougrday** [comment]
This was released to NuGet in version 0.2.1-beta:  https://www.nuget.org/packages/Dapper.GraphQL/0.2.1-beta

**Comment by benmccallum** [comment]
Interesting, it passed for me when I ran it alone. But I did get one failing when I "ran all" the existing tests together before adding the new test.

Is this because this IDbConnection is being shared once per test run and a previous execution will close the connection so it needs to be re-opened? Otherwise I'd expect to see connection.Open() in all  the resolvers before .Execute() or .ExecuteAsync()

If so, should the lifetime of the IDbConnection in DI be changed?

**Comment by dougrday** [comment]
It's because the DbConnection that is used currently isn't thread-safe, so it's getting mangled by all the tests.

Plus, the unit tests are terribly slow because of the LocalDB design, I'm going to try some options to speed it up and avoid these issues altogether.

**Comment by benmccallum** [comment]
I see, that makes sense, thanks. 

I've been wondering myself about how best to manage the IDbConnection in my own asp.net core app. I see in the resolvers you're using the serviceProvider injected in the Query object, and so I should probably just configure a transient IDbConnection in my Startup like so:

```
services.AddTransient<IDbConnection>(
    x => new SqlConnection(Config.GetConnectionString("SqlDb"))
);
```

**Comment by benmccallum** [comment]
Hey @dougrday, Do you think you could take a sec to release a new version with the rest of the async methods I added in there? Looking at using them now. Thanks mate! You can prob close this out then too.

**Comment by dougrday** [comment]
Remaining features were released to NuGet in version 0.2.2-beta: https://www.nuget.org/packages/Dapper.GraphQL/0.2.2-beta

**Comment by benmccallum** [comment]
Thanks Doug!
